pkg/virt-controller/watch/vmi.go

```diff
@@ -242,11 +244,17 @@ func (c *VMIController) runWorker() {
 	}
 }
 
+var virtControllerVMIWorkQueueTracer = &traceUtils.Tracer{Threshold: time.Second}
+
 func (c *VMIController) Execute() bool {
 	key, quit := c.Queue.Get()
 	if quit {
 		return false
 	}
+
+	virtControllerVMIWorkQueueTracer.StartTrace(key.(string), "virt-controller VMI workqueue", trace.Field{Key: "Workqueue Key", Value: key})
+	defer virtControllerVMIWorkQueueTracer.StopTrace(key.(string))
+
 	defer c.Queue.Done(key)
 	err := c.execute(key.(string))
```

Comment:
You have created _steps_ for the `updateStatus` and `sync` functions, which are called within the `execute` function.

So, if there is no strong reason to create a new library as I mentioned before, all _steps_ could go into the `execute` function.

Reply:
I want to use the pattern of adding a `defer Step` at the start of a function we want to trace in order to avoid any accidents where we don't end up recording the code we want.